lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -44,26 +50,31 @@ def deploy_succeeded?
     end
 
     def deploy_failed?
-      @latest_rs && @latest_rs.deploy_failed?
+      @latest_rs&.deploy_failed?
     end
 
     def failure_message
-      @latest_rs&.failure_message
+      return unless @latest_rs.present?
+      "Latest ReplicaSet: #{@latest_rs.name}\n\n#{@latest_rs.failure_message}"
     end
 
     def timeout_message
-      progress_seconds = @definition['spec']['progressDeadlineSeconds']
-      if progress_seconds
-        "Deploy timed out due to progressDeadlineSeconds of #{progress_seconds} seconds, "\
-        " reason: #{@progress['reason']}\n"\
-        "#{@latest_rs&.timeout_message}"
+      reason_msg = if @progress_condition.present?
+        "Timeout reason: #{@progress_condition['reason']}"
       else
-        @latest_rs&.timeout_message
+        "Timeout reason: hard deadline for #{type}"
       end
+      return reason_msg unless @latest_rs.present?
+      "#{reason_msg}\nLatest ReplicaSet: #{@latest_rs.name}\n\n#{@latest_rs.timeout_message}"
+    end
+
+    def pretty_timeout_type
+      @progress_deadline.present? ? "progress deadline: #{@progress_deadline}s" : super
```

Comment:
When will `progress_deadline?` not be present?

We require its presence in https://github.com/Shopify/kubernetes-deploy/pull/213/files#diff-4709b5257970978173e422904663525fR92

Reply:
There is no server-side default in `extensions/v1beta1`, so it will not be present in an `extensions/v1beta1` deployment that doesn't explicitly include the field. I have a test that uses such a deployment, actually: 

![image](https://user-images.githubusercontent.com/4789493/33409330-db31b1f8-d548-11e7-9a98-0ec8066086a2.png)

The method you linked actually has a `return false unless @progress_condition.present?` guard, though there's a giant comment making that hard to notice. 😄